### PR TITLE
Fix green esco skills

### LINKED ISSUE #110: Green ESCO skills not being utilised in full
In `green_esco_formatting.py` we join the green ESCO skills to the dataset created last year of ESCO skills to where they sit in the taxonomy. This step is actually unnecessary since for this work we don't use the taxonomy information anyway, also it is out of date, so it means the 570 green ESCO skills (preferred label) are filtered down to 427 - that's 4326 to 3513 when you inc preferred plus altlabels!

It means we aren't mapping to a bunch of skills we should be, including 'perform feasibility study on solar heating', 'apply procedures and regulations for eco-labelling', 'implement sustainable procurement', 'ecotourism', 'energy storage systems'!

Since this is quite a big chunk of green ESCO skills I think we need to fix it!

- [x] Fix `green_esco_formatting.py`
- [x] Recreate `outputs/data/green_skill_lists/green_esco_data_formatted_{date_stamp}.csv` and `outputs/data/green_skill_lists/green_esco_embeddings_{date_stamp}.json`
- [x] Update mentions of these files everywhere
- [x] Retrain greenskills classifier
- [ ] Rerun skills measures
- [ ] Think about how this changes our skills evaluation? I don't see how our results should be worse from this as we are only adding to the options to map to, rather than decreasing.

### PR DESCRIPTION
**Pull request by lizgzil**
---

# Description

Please include a summary of the changes.

Fixes #110 

Analysis shows sometimes the new ESCO green skills have slightly different names.

Old ESCO method: 3513 green skills (altlabels + preferred labels), 427 unique IDs.
New ESCO method: 4288 green skills (altlabels + preferred labels), 570 unique IDs.

- 0 IDs from the old method that weren't in the new
- 12 times where the old description is not in any of the new descriptions (updates to labels I suppose)
- e.g. 'handle the disposal of chemicals' used to be a description, but now the closest is "handle the disposal of chemicals correctly". All of the 12 are differences like this (i.e. very minor)
- 
After deduplicating repeated descriptions (a step which we didn't do last time - but if we had we'd have 3482). We have 3869 green skill descriptions from 570 unique green skills.

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

**Comment by lizgzil** [comment]
The full 19 times an ESCO green skill has the same preferred or altlabel but different ids are below. The rule we follow is to keep the "skill/competence" one (if possible) otherwise just pick the first one.

manage waste
['73a9561d-112c-4d8a-9d5f-b2c11c5bbd5f', '40f65a56-ccbe-4601-9f32-1cc6cdd24f28']


waste managing
['40f65a56-ccbe-4601-9f32-1cc6cdd24f28', '73a9561d-112c-4d8a-9d5f-b2c11c5bbd5f']


advocate sustainability
['469e19ed-a0bd-445a-ae2d-4ba9430e296b', '7f321b88-3d16-4e30-8163-5494ca8ce978']


analyse energy consumption
['29bfe2cd-f011-4f94-a3aa-f77768e60460', '0ad6d5e8-593f-4191-843c-439b23a468ca']


manage environmental impact
['d3a8ef9c-8572-438c-91dc-010537070be3', 'd86babea-bf8b-4552-81a8-daf706066ad6']


environmental legislation
['6f5850ea-5785-4e2c-b062-743066fd5bdb', 'c1e0de6d-ec3b-4323-92d4-d1c62df2064a']


environmental impact management
['d3a8ef9c-8572-438c-91dc-010537070be3', 'd86babea-bf8b-4552-81a8-daf706066ad6']


biomass conversion
['b115cabc-c030-45b2-ac8e-922d45716183', '1b718a43-4295-44cc-9530-78323bbeb663']


International Convention for the Prevention of Pollution from Ships
['aac9e93a-106a-4cdc-8d27-ac3a9ffce0ac', '75f312e1-622b-4751-8d93-815b32feaf8d']


energy consumption analysing
['0ad6d5e8-593f-4191-843c-439b23a468ca', '29bfe2cd-f011-4f94-a3aa-f77768e60460']


analysing energy consumption
['0ad6d5e8-593f-4191-843c-439b23a468ca', '29bfe2cd-f011-4f94-a3aa-f77768e60460']


dispose of waste
['5e433def-210d-4dfa-bf07-dbf6d04e0eaa', '73a9561d-112c-4d8a-9d5f-b2c11c5bbd5f']


follow recycling collection schedules
['ca4ae285-244f-4321-83d6-b109bd1693bc', '4c3ab565-747c-4db3-8c47-79e8d6f98cf4']


managing waste
['40f65a56-ccbe-4601-9f32-1cc6cdd24f28', '73a9561d-112c-4d8a-9d5f-b2c11c5bbd5f']


conversion of biomass
['1b718a43-4295-44cc-9530-78323bbeb663', 'b115cabc-c030-45b2-ac8e-922d45716183']


MARPOL
['75f312e1-622b-4751-8d93-815b32feaf8d', 'aac9e93a-106a-4cdc-8d27-ac3a9ffce0ac']


promote sustainability
['469e19ed-a0bd-445a-ae2d-4ba9430e296b', '7f321b88-3d16-4e30-8163-5494ca8ce978']


principles of ecology
['62685d7a-a8be-4070-89e9-8699ad1201b6', 'a8429098-ddac-481b-91e7-688d021c092e']


ecological principles
['62685d7a-a8be-4070-89e9-8699ad1201b6', 'a8429098-ddac-481b-91e7-688d021c092e']

File: dap_prinz_green_jobs/config/base.yaml [hunk 1]
```diff
@@ -20,8 +20,8 @@ skills:
   load_taxonomy_embeddings: True # If False it doesn't matter what green_tax_embedding_path is
   skills_output: "outputs/data/green_skill_lists/20230914/skills_data_ojo_mixed.json" # Set if load_skills is True
   skill_embeddings_output: "outputs/data/green_skill_lists/20230914/extracted_skills_embeddings.json" # Set if load_skills_embeddings is True
-  green_tax_embedding_path: "outputs/data/green_skill_lists/green_esco_embeddings_20230815.json" # Set if load_taxonomy_embeddings is True
-  green_skills_classifier_model_file_name: "outputs/models/green_skill_classifier/green_skill_classifier_20230906.joblib"
```
Comment (india-kerle) [inline dap_prinz_green_jobs/config/base.yaml:24]: I reckon we can just get rid of these guys in s3 given they're based off of a bug in esco

File: dap_prinz_green_jobs/config/extract_green_skills_esco.yaml [hunk 2]
```diff
@@ -1,9 +1,9 @@
 ner_model_path: "outputs/models/ner_model/20230808/"
 taxonomy_name: "green_esco"
-taxonomy_path: "outputs/data/green_skill_lists/green_esco_data_formatted.csv"
```
Comment (india-kerle) [inline dap_prinz_green_jobs/config/extract_green_skills_esco.yaml:3]: ditto to delete in s3

File: dap_prinz_green_jobs/pipeline/green_measures/skills/green_esco_formatting.py [hunk 3]
```diff
@@ -1,43 +1,109 @@
...
+    logger.info("Formatting the esco green skills taxonomy")
+
+    esco_green_skills["type"] = "preferredLabel"
+
+    # Expand out the alternate labels
+    esco_green_skills["altLabels_expanded"] = esco_green_skills["altLabels"].apply(
+        lambda x: x.split("|") if isinstance(x, str) else None
     )
+    esco_green_skills_exploded = esco_green_skills.explode("altLabels_expanded")
+    esco_green_skills_exploded["type"] = "altLabel"
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/green_esco_formatting.py:40]: omg there it is - such a good catch frankly

File: dap_prinz_green_jobs/pipeline/green_measures/skills/green_esco_formatting.py [hunk 4]
```diff
@@ -1,43 +1,109 @@
...
+            lambda x: x
+            if "preferredLabel" not in x["type"].tolist()
+            else x.loc[x.type.eq("preferredLabel")]
+        )
+        .reset_index(drop=True)
+    )
+
+    # There are times when the description is the same but the id isn't, we need to just have one, so keep the one
+    # from skillType = "skill/competence" (rather than "knowledge") if possible otherwise pick the first one.
+    # Note this only happens 19 times
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/skills/green_esco_formatting.py:76]: wow this is a really good catch - feels like we should feed back to esco eventually that not all skills have a unique id associated to them!

**Review by india-kerle — APPROVED** [review]
truly such a good catch - i reckon its worth deleting the old model / embeddings but otherwise i think its good to merge 😎

**Pull request merged**